packages/shorebird_code_push_client/test/src/code_push_client_test.dart

```diff
@@ -107,50 +101,62 @@ void main() {
       });
 
       test('completes when request succeeds', () async {
-        when(
-          () => httpClient.post(
-            any(),
-            headers: any(named: 'headers'),
-            body: any(named: 'body'),
+        when(() => httpClient.send(any())).thenAnswer(
+          (_) async => http.StreamedResponse(
+            const Stream.empty(),
+            HttpStatus.created,
           ),
-        ).thenAnswer((_) async => http.Response('', HttpStatus.created));
+        );
 
         await codePushClient.createCollaborator(appId: appId, email: email);
 
-        final uri = verify(
-          () => httpClient.post(
+        final request = verify(
+          () => httpClient.send(
             captureAny(),
-            headers: any(named: 'headers'),
-            body: any(named: 'body'),
           ),
-        ).captured.single as Uri;
+        ).captured.single as http.BaseRequest;
 
         expect(
-          uri,
+          request.url,
           codePushClient.hostedUri.replace(
             path: '/api/v1/apps/$appId/collaborators',
           ),
         );
+        expect(
+          CodePushClient.headers.entries.every(
+            (entry) => request.headers[entry.key] == entry.value,
+          ),
+          isTrue,
+        );
```

Comment:
Should we be checking http method here now that we're mocking `send` instead of the http verb methods?

Reply:
Yup will do 👍